android/src/main/cpp/FrameProcessorRuntimeManager.cpp

```diff
@@ -105,16 +95,18 @@ void FrameProcessorRuntimeManager::logErrorToJS(const std::string& message) {
   });
 }
 
-void FrameProcessorRuntimeManager::setFrameProcessor(jsi::Runtime& runtime,
+void FrameProcessorRuntimeManager::setFrameProcessor(jsi::Runtime& rnRuntime,
                                                      int viewTag,
-                                                     const jsi::Value& frameProcessor) {
+                                                     const jsi::Value& frameProcessor,
+                                                     const jsi::Value& workletRuntimeValue) {
   __android_log_write(ANDROID_LOG_INFO, TAG,
                       "Setting new Frame Processor...");
 
-  if (!_runtimeManager || !_runtimeManager->runtime) {
-    throw jsi::JSError(runtime,
-                       "setFrameProcessor(..): VisionCamera's RuntimeManager is not yet initialized!");
-  }
+  workletRuntime_ = reanimated::extractWorkletRuntime(rnRuntime, workletRuntimeValue);
```

Comment:
We create the worklet runtime in JavaScript using `createWorkletRuntime`, then pass it to C++ (since it's a jsi::HostObject) and here's how we extract the underlying C++ class `WorkletRuntime`.